packages/terra-application-header/src/ApplicationHeader.jsx

```diff
@@ -0,0 +1,177 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import AppDelegate from 'terra-app-delegate';
+import RoutingStackDelegate from 'terra-navigation-layout/lib/RoutingStackDelegate';
+import ApplicationHeaderLayout from 'terra-application-header-layout';
+// import { ApplicationHeaderUtility } from 'terra-application-utility';
+// import { ApplicationHeaderName } from 'terra-application-name';
+// import { ApplicationTabs } from 'terra-application-links';
+import IconMenu from 'terra-icon/lib/icon/IconMenu';
+import Button from 'terra-button';
+import Popup from 'terra-popup';
+
+import 'terra-base/lib/baseStyles';
+
+import ApplicationHeaderUtility from './mock-components/MockApplicationUtility';
+import ApplicationHeaderName from './mock-components/MockApplicationName';
+import ApplicationTabs from './mock-components/MockApplicationLinks';
+
+import styles from './ApplicationHeader.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * The AppDelegate instance that will be propagated to the components presented within the NavigationLayout.
+   */
+  app: AppDelegate.propType,
+  /**
+   * Navigational links that will generate list items that will update the path. These paths are matched with react-router to selection.
++  */
+  applicationLinks: PropTypes.arrayOf(PropTypes.shape({
+    id: PropTypes.string,
+    path: PropTypes.string.isRequired,
+    text: PropTypes.string.isRequired,
+  })),
+  /**
+   * Configutation values for the ApplicationName component.
+   */
+  nameConfig: PropTypes.shape({
+    accessory: PropTypes.element,
+    title: PropTypes.string,
+  }),
+  /**
+   * Layout config provided from the Layout component.
+   */
+  layoutConfig: PropTypes.shape({
+    size: PropTypes.string,
+    toggleMenu: PropTypes.func,
+    menuIsOpen: PropTypes.bool,
+    togglePin: PropTypes.bool,
+    menuIsPinned: PropTypes.bool,
+  }).isRequired,
+  /**
+   * Delegate prop that is padded by the NavigationLayout.
+   */
+  routingStackDelegate: RoutingStackDelegate.propType,
+  /**
+   * Configration to be provided to the ApplicationUtility component.
+   */
+  utilityConfig: PropTypes.shape({
+    userName: PropTypes.string,
+    userPhoto: PropTypes.element,
+    userDetails: PropTypes.string,
+    onUtilityChange: PropTypes.func,
+  }).isRequired,
+};
+
+const defaultProps = {
+  applicationLinks: [],
+};
+
+class ApplicationHeader extends React.Component {
+  constructor(props) {
+    super(props);
+    this.onDiscloseUtilty = this.onDiscloseUtilty.bind(this);
+    this.handleRequestClose = this.handleRequestClose.bind(this);
+    this.getTargetRef = this.getTargetRef.bind(this);
+    this.setContentNode = this.setContentNode.bind(this);
+    this.state = { isUtilityOpen: false };
+  }
+
+  onDiscloseUtilty(utility) {
+    if (utility && !this.state.isUtilityOpen) {
+      this.popupContent = utility;
```

Comment:
Can you keep the utility component stored in state directly and lose the open flag? From what I can tell you're keeping them in sync at the moment anyway.